Update to 3.9.0

JamesRobertsonGames opened the pull request.

JamesRobertsonGames committed:
remove old readme info

JamesRobertsonGames committed:
add formatting

JamesRobertsonGames committed:
revert formatting

JamesRobertsonGames committed:
add formatting

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:53: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:68: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
WARNING: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:68: missing_description: The recipe is missing a description
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:71: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:71: missing_description: The recipe is missing a description
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_18oxs7e0ev/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
Errors were found

</details>

JamesRobertsonGames committed:
remove the liblapack and libblas dependencies

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:66: missing_description: The recipe is missing a description
ERROR: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:66: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:66: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:66: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:69: missing_description: The recipe is missing a description
ERROR: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:69: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_3bhrdicnzv/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
Errors were found

</details>

JamesRobertsonGames committed:
remove libblas with openblas

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:67: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:67: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:67: missing_description: The recipe is missing a description
ERROR: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:67: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:70: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:70: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:70: missing_dev_url: The recipe is missing a dev_url
WARNING: /tmp/abs_e9nmyziqqe/clone/recipe/meta.yaml:70: missing_description: The recipe is missing a description
Errors were found

</details>

JamesRobertsonGames committed:
change to a OPENBlas with implied openblas

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:67: missing_description: The recipe is missing a description
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:67: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:67: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:67: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
WARNING: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:69: missing_description: The recipe is missing a description
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:69: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:70: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:70: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:70: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_40cn1qn0ab/clone/recipe/meta.yaml:70: missing_description: The recipe is missing a description
Errors were found

</details>

JamesRobertsonGames committed:
remove for overlinking

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:66: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:66: missing_description: The recipe is missing a description
ERROR: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:66: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:66: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:69: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_f55a0p8pc5/clone/recipe/meta.yaml:69: missing_description: The recipe is missing a description
Errors were found

</details>

JamesRobertsonGames committed:
remove faulty testing scheme

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:66: missing_description: The recipe is missing a description
ERROR: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:66: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:66: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:66: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:69: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:69: missing_description: The recipe is missing a description
ERROR: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_360x9pi2ck/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
Errors were found

</details>

JamesRobertsonGames committed:
change the method of testing

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:0: missing_tests: No tests were found.
ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:56: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:56: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:56: missing_description: The recipe is missing a description
ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:56: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:59: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:59: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:59: missing_description: The recipe is missing a description
ERROR: /tmp/abs_74xzw33r6f/clone/recipe/meta.yaml:59: missing_documentation: The recipe is missing a doc_url or doc_source_url
Errors were found

</details>

JamesRobertsonGames committed:
revert build changes to failed tests

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:66: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:66: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:66: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:66: missing_description: The recipe is missing a description
ERROR: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:69: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_a6aww8prjt/clone/recipe/meta.yaml:69: missing_description: The recipe is missing a description
Errors were found

</details>

akabanovs committed:
Try adding openblas

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:68: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:68: missing_description: The recipe is missing a description
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:69: missing_description: The recipe is missing a description
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:69: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:71: missing_description: The recipe is missing a description
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_caja0c2zyb/clone/recipe/meta.yaml:71: missing_license_family: The recipe is missing the `about/license_family` key.
Errors were found

</details>

sumit0190 commented:
This fails to build for two different reasons:
1) on all architectures except `win-64`, you are specifying `openblas` as a host dependency only when `blas_impl` equals `openblas`, but when the build kicks off the first time it's not equal to `openblas` - it's set to `mkl` (it iterates over both the values). Naturally, this means that no BLAS and LAPACK libraries are found and the tests fail (all of them). To solve this, you either remove the selector OR you add another selector specifying the `mkl` libraries when `blas_impl` is set to `mkl`. But beware that you'd have to set a CMake flag in order to let it find and work with the MKL libraries. To simplify things, you could probably skip the MKL variant and just work with `openblas` for everything except `win-64`. I verified on my Linux machine that this works and the package gets built.
2) on `win-64`, `openblas` doesn't exist. So you have to either make it work with MKL (again, by specifying the right CMake flag - it's in the documentation I think), or skip this architecture altogether.

akabanovs committed:
Removing blas selector and change -DBLAS_LIBRARIES flag to cmake

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:68: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:68: missing_description: The recipe is missing a description
ERROR: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:71: missing_description: The recipe is missing a description
ERROR: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:71: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_a9tz69l865/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
Errors were found

</details>

akabanovs committed:
s390x does not have openmpi

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:68: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:68: missing_description: The recipe is missing a description
ERROR: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
WARNING: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:71: missing_description: The recipe is missing a description
ERROR: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:71: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_76xim43qg3/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
Errors were found

</details>

akabanovs committed:
Further attempts:
 - openmpi conflict on linux-64 - switching to mpich
 - unable to locate libquadmath.so.0, so explicitly passing the path in build.sh

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
WARNING: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:68: missing_description: The recipe is missing a description
ERROR: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:68: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:71: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:71: missing_description: The recipe is missing a description
ERROR: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_f8xyqx1p24/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
Errors were found

</details>

akabanovs committed:
Remove openmpi altogether

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
WARNING: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:68: missing_description: The recipe is missing a description
ERROR: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:68: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:71: missing_description: The recipe is missing a description
ERROR: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:71: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_fafu8w1nrh/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
Errors were found

</details>

akabanovs committed:
Try adding -fallow-argument-mismatch as a cmake argument

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:48: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:68: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:68: missing_description: The recipe is missing a description
ERROR: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:71: missing_description: The recipe is missing a description
ERROR: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:71: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_cabytnfav9/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
Errors were found

</details>

akabanovs committed:
Build v 3.9.0; Allow to use mkl; fix C/CPP flags on osx

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:70: missing_dev_url: The recipe is missing a dev_url
WARNING: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:70: missing_description: The recipe is missing a description
ERROR: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:70: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:70: missing_license_family: The recipe is missing the `about/license_family` key.
ERROR: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:74: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:74: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:74: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_cb88me8993/clone/recipe/meta.yaml:74: missing_description: The recipe is missing a description
Errors were found

</details>

akabanovs committed:
Add if clause

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:70: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:70: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:70: missing_description: The recipe is missing a description
ERROR: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:70: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:74: missing_documentation: The recipe is missing a doc_url or doc_source_url
WARNING: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:74: missing_description: The recipe is missing a description
ERROR: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:74: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_47ezqh9rj9/clone/recipe/meta.yaml:74: missing_license_family: The recipe is missing the `about/license_family` key.
Errors were found

</details>

akabanovs committed:
Revert back to v 3.8.0 with no mkl support

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:50: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:70: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:70: missing_license_family: The recipe is missing the `about/license_family` key.
WARNING: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:70: missing_description: The recipe is missing a description
ERROR: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:70: missing_dev_url: The recipe is missing a dev_url
WARNING: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:75: missing_description: The recipe is missing a description
ERROR: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:75: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:75: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_62dal4mquu/clone/recipe/meta.yaml:75: missing_license_family: The recipe is missing the `about/license_family` key.
Errors were found

</details>

akabanovs committed:
Set version to 3.9.0; Remove mkl support; tidy up

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_2f1lh9vgvt/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Build with openmpi only on osx-arm64

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_ba2yguzmer/clone/recipe/meta.yaml:42: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

JeanChristopheMorinPerso commented on recipe/build.sh:
```diff
@@ -59,5 +41,8 @@ do
 
 if [[ "${CONDA_BUILD_CROSS_COMPILATION}" != "1" ]]; then
-  ctest --output-on-failure -j${CPU_COUNT}
+  # Do not run ctest on osx as it causes our ci to crash.
```
I would make it extra clear that the OS is crashing.

sumit0190 replied:
Might just be best to say "this test causes crashes on certain osx configurations".

JeanChristopheMorinPerso commented on recipe/conda_build_config.yaml:
```diff
@@ -1,4 +1,4 @@
-  - mpich    # [not win and not (osx and arm64)]
-  - openmpi  # [not win and not (osx and arm64)]
+  - nompi 
+  - mpich  # [not win]
```
Should we pin `mpich` and `openmpi`? It might also be interesting to document why one is needed on windows while the other is needed on the other platforms.

akabanovs replied:
The stuff there is used as a matrix so artefacts are built for each separate entry. We just only use nompi for windows (mpich and openmpi not available/supported on windows I think?). 
The reason why we do not build with openmpi on anything apart for osx-arm64 is because there seem to be some dependency conflict on all archs except for osx-arm64. The moment I switch from openmpi to mpich - the conflict goes away... A separate ticket will be created to investigate. It seems that the igraph is pretty urgent, and arpack is its dependency. Therefore I thought we could investigate the issue with openmpi later. @sumit0190  would be good to hear your opinion on this too, please.

sumit0190 replied:
For now we should just document why `openmpi` isn't being used on anything else other than `osx-arm64`. The `windows` part probably doesn't need any clarification since no MPI implementation other than the one offered by MS is available.

sumit0190 commented on recipe/meta.yaml:
```diff
@@ -1,31 +1,21 @@
 
 build:
   number: {{ build }}
+  skip: true  # [blas_impl == "mkl"]
```
Could you either remove the references to `blas_impl` (and therefore `mkl`) entirely, or add a comment saying we are temporarily skipping MKL builds?

akabanovs replied:
Currently does not build for mkl. If I remove skip, it will try to build for it and will fail. Therefore, gonna add a comment.

akabanovs replied:
no, you are right, can simply remove blas_impl reference and it should not try to build for mkl which is contained within blas_impl matrix.

sumit0190 reviewed (approved):
LGTM apart from that one comment about removing the reference to `blas_impl`.

akabanovs committed:
Remove reference to blas_impl and improve a comment about tests

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_d1pv8kxdh6/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
should not use openblas with nompi

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_091_erbe03/clone/recipe/meta.yaml:41: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Try to build with mkl on windows

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_05f5wlze4a/clone/recipe/meta.yaml:46: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Try to build with mkl on windows attempt 2.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_2a81reu0s5/clone/recipe/meta.yaml:46: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Try to build with mkl on windows attempt 3.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_d393wdvmr7/clone/recipe/meta.yaml:46: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Try to build with mkl on windows attempt 4.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
             ^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
               ^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 748, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/__init__.py", line 247, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.11/site-packages/anaconda_linter/lint/check_syntax.py", line 36, in check_recipe
    has_constraints = constraints.search(spec)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: expected string or bytes-like object, got 'NoneType'

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_41_j7mkqbt/clone` failed. (See above for error)
</details>

akabanovs committed:
Fix typo

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_b2ebmz9h98/clone/recipe/meta.yaml:46: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
cleanup and add DMIP=OFF flag to bld.bat

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_52rp200zsh/clone/recipe/meta.yaml:45: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Remove cmake flag

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_63mlfhcj4t/clone/recipe/meta.yaml:45: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Remove cmake flag

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_9chk4to3m1/clone/recipe/meta.yaml:45: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs commented:
diff against previously reviewed variant:

https://github.com/AnacondaRecipes/arpack-feedstock/pull/1/files/cbfb5c1e33d5e3970d8c6084743af6b24f9703b1..0e818f81ff0e9edcdc445e2c4740d8b149db4d29

JeanChristopheMorinPerso commented:
The tests are badly behaving on linux-ppc64le. It makes the prefect agent hang (not crash).

Also I see that the package is being built twice on Linux, once with
```
_libgcc_mutex:  0.1-main          defaults
_openmp_mutex:  5.1-1_gnu         defaults
blas:           1.0-openblas      defaults
libgcc-ng:      11.2.0-h1234567_1 defaults
libgfortran-ng: 11.2.0-hb3889a9_1 defaults
libgfortran5:   11.2.0-h1234567_1 defaults
libgomp:        11.2.0-h1234567_1 defaults
libopenblas:    0.3.21-hd3ce00b_0 defaults
libstdcxx-ng:   11.2.0-h1234567_1 defaults
nomkl:          3.0-0             defaults
openblas:       0.3.21-h6ffa863_0 defaults
openblas-devel: 0.3.21-h6ffa863_0 defaults
```
and once with
```
_libgcc_mutex:  0.1-main          defaults
_openmp_mutex:  5.1-1_gnu         defaults
blas:           1.0-openblas      defaults
libgcc-ng:      11.2.0-h1234567_1 defaults
libgfortran-ng: 11.2.0-hb3889a9_1 defaults
libgfortran5:   11.2.0-h1234567_1 defaults
libgomp:        11.2.0-h1234567_1 defaults
libopenblas:    0.3.21-hd3ce00b_0 defaults
libstdcxx-ng:   11.2.0-h1234567_1 defaults
mpi:            1.0-mpich         defaults
mpich:          4.1.1-h0a38bd9_0  defaults
nomkl:          3.0-0             defaults
openblas:       0.3.21-h6ffa863_0 defaults
openblas-devel: 0.3.21-h6ffa863_0 defaults
```
Is this expected?

akabanovs commented:
> The tests are badly behaving on linux-ppc64le. It makes the prefect agent hang (not crash).
> 
> Also I see that the package is being built twice on Linux, once with
> 
> ```
> _libgcc_mutex:  0.1-main          defaults
> _openmp_mutex:  5.1-1_gnu         defaults
> blas:           1.0-openblas      defaults
> libgcc-ng:      11.2.0-h1234567_1 defaults
> libgfortran-ng: 11.2.0-hb3889a9_1 defaults
> libgfortran5:   11.2.0-h1234567_1 defaults
> libgomp:        11.2.0-h1234567_1 defaults
> libopenblas:    0.3.21-hd3ce00b_0 defaults
> libstdcxx-ng:   11.2.0-h1234567_1 defaults
> nomkl:          3.0-0             defaults
> openblas:       0.3.21-h6ffa863_0 defaults
> openblas-devel: 0.3.21-h6ffa863_0 defaults
> ```
> 
> and once with
> 
> ```
> _libgcc_mutex:  0.1-main          defaults
> _openmp_mutex:  5.1-1_gnu         defaults
> blas:           1.0-openblas      defaults
> libgcc-ng:      11.2.0-h1234567_1 defaults
> libgfortran-ng: 11.2.0-hb3889a9_1 defaults
> libgfortran5:   11.2.0-h1234567_1 defaults
> libgomp:        11.2.0-h1234567_1 defaults
> libopenblas:    0.3.21-hd3ce00b_0 defaults
> libstdcxx-ng:   11.2.0-h1234567_1 defaults
> mpi:            1.0-mpich         defaults
> mpich:          4.1.1-h0a38bd9_0  defaults
> nomkl:          3.0-0             defaults
> openblas:       0.3.21-h6ffa863_0 defaults
> openblas-devel: 0.3.21-h6ffa863_0 defaults
> ```
> 
> Is this expected?

Not sure about tests. But I also noticed that last time it took 5 minutes to build on linux-64, and 5 hours on linux-ppc64le... 

Regarding the package and it being built twice - that is fine. It builds nompi variant, and mpi variant that uses mpich mpi library. So there should be 2 artefacts generated.

JeanChristopheMorinPerso commented:
https://cloud.prefect.io/flow-run/cc981ccf-1835-4906-9e24-2bf044c99651 has been running for 5 hours. I don't think that it is fair for a job to run tests for 5 hours. Something should be done to avoid this.

akabanovs commented:
> Member

Yes, I agree. I only now noticed that.

akabanovs committed:
Test: deactivate nompi builds on non-windows

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_d6h6ccnn1q/clone/recipe/meta.yaml:45: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
hardcode the number of cores to use in tests

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_22tdqngrzl/clone/recipe/meta.yaml:45: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs commented:
I am not sure what to do about linux-ppc64le build. I don't think the tests are hanging. My guess is that the number of allocated processes is not enough to compute tests quickly enough, the tests need sufficient resources, plus tests on this arch run generally slower for some reason. When I compare the number of allocated processes, linux-ppc64le runs something like 2-3 processes concurrently, while s390x, which runs much quicker, utilises 17. I tried to manually specify -j16 as an argument to ctest, but it did not make any difference and I can still see CPU usage of ~120% if I ssh into the machine and monitor it. At the same time, if I build locally on linux-ppc64le the build uses 128 CPU and CPU usage spins to 1000+ %. It takes some time to build and run tests, but not 5h. I am not sure if the problem is in our hardware, docker/podman image, or the package and the contents of the recipe... 

I know that this package is urgent. In the worst case, we could just deactivate linux-ppc64le. Otherwise we could deactivate tests on this arch to make it produce artifacts since it seems to work fine if not run on prefect.  What do you think @sumit0190 @JeanChristopheMorinPerso we should do. Maybe any of you can have another look?

akabanovs committed:
Skip linux-ppc64le

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_2dpo22mb2d/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

JeanChristopheMorinPerso commented:
I don't know what should be done. The only thing I know is that we should be careful to not block other builds from running for too long (like 5 hours) just to run tests.

akabanovs commented:
Okay, based on @sumit0190  suggestion, I am skipping ppc arch altogether. It is worth mentioning, however, that linux-aarch64 is also taking much time to build ~ 1h  17 min, tests taking ~51 minute. linux-64 finishes building and testing in ~15 minutes.  osx skip tests so they build almost in no time.

JeanChristopheMorinPerso commented:
Thanks @akabanovs.

>It is worth mentioning, however, that linux-aarch64 is also taking much time to build ~ 1h 17 min, tests taking ~51 minute.

Thanks for mentioning that. I think it's fine because we have more linux-4 and linux-aarch64 workers than on other platforms and archs.

akabanovs commented:
Interestingly, the last build on linux-aarch64 took:

Total time: 0:13:19.1
Resource usage statistics from testing arpack:
Process count: 19

And the one before:

Total time: 1:17:33.0
Resource usage statistics from testing arpack:
Process count: 2

akabanovs committed:
Upload to staging channel so that we can start working on the next package while waiting for review of this one

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_0e84hx00bc/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

M-Waszkiewicz-Anaconda commented on recipe/build.sh:
```diff
@@ -8,21 +8,41 @@ if [[ "$(echo $fortran_compiler_version | cut -d '.' -f 1)" -gt 9 ]]; then
 
 if [[ "${CONDA_BUILD_CROSS_COMPILATION}" != "1" ]]; then
-  ctest --output-on-failure -j${CPU_COUNT}
+  # Do not run the test on osx as this test causes OS crashes on certain osx configurations.
```
does it mean for specific osx configuration  the code could crash on customer devices ? if so then better to exclude those configuration from the build itself, better not have the build at all than the invalid one which upsets customers

skupr-anaconda reviewed (approved):
LGTM but the MPI/OpenMP/OpenBLAS are fozzy for me

M-Waszkiewicz-Anaconda reviewed (approved):
LGTM

skupr-anaconda commented on recipe/build.sh:
```diff
@@ -8,21 +8,41 @@ if [[ "$(echo $fortran_compiler_version | cut -d '.' -f 1)" -gt 9 ]]; then
-  ctest --output-on-failure -j${CPU_COUNT}
+  # Do not run the test on osx as this test causes OS crashes on certain osx configurations.
+  if [[ ${HOST} =~ .*linux.* ]]; then
+    ctest --output-on-failure -j${CPU_COUNT}
```
I remember that Kai said not to use multiprocessing  `-j${CPU_COUNT` on `ppc64le`. Might it be the root cause of issues on that platform?

akabanovs replied:
Thanks. That might explain it.

akabanovs committed:
Remove abs.yaml file

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_54ht9wrmzl/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

skupr-anaconda commented on recipe/build.sh:
```diff
@@ -8,21 +8,41 @@ if [[ "$(echo $fortran_compiler_version | cut -d '.' -f 1)" -gt 9 ]]; then
-    -DBLAS_LIBRARIES="-lblas" \
     -DICB=ON \
     -DMPI=${DMPI} \
+    -DBLAS_LIBRARIES="-lopenblas" \
```
Do we need `-DEIGEN=ON -DPYTHON3=ON` too? See https://github.com/opencollab/arpack-ng/blob/6d625dbf34375a0b5823d6c03bd16044f405baae/.github/workflows/jobs.yml#LL132C49-L132C72

akabanovs replied:
To allow python3 support through -DPYTHON3 flag, arpack asks for boost v 1.78 but our latest is 1.73.

skupr-anaconda replied:
Ah, thanks

akabanovs replied:
I have added EIGEN support.

akabanovs committed:
Try rebuilding with EIGEN support

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_7cybyg_jty/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_7cybyg_jty/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_7cybyg_jty/clone/recipe/meta.yaml:53: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Adding abs.yaml

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_e1t_0f0wg1/clone/recipe/meta.yaml:49: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e1t_0f0wg1/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_e1t_0f0wg1/clone/recipe/meta.yaml:53: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Remove abs.yaml and add pinning to eigen

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_23t954mhfe/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Use eigen on non-windows only

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_81msegw95v/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Another attempt to build with Eigen support on win

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_f02glo18sa/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Revert changes as eigen flag passed to CMake project is not used.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_5ay3efo_8w/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

JeanChristopheMorinPerso reviewed (commented):
Looks good to me, but it's far from my expertise, so I won't approve since I would consider my approval an "empty" approval. But since @sumit0190 approved, then you are good to go.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_d9m5lik9ik/clone/recipe/meta.yaml:51: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>
